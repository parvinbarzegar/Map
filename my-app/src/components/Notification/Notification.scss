.notification {
    min-width: 250px;
    max-width: 700px;
    z-index: 99999999999999;
    height: 60px;
    display: flex;
    align-items: center;
    justify-content: flex-start;
    padding-left: 1em;
    padding-right: 1em;
    border-radius: 10px;
    color: white;
    position: absolute;
    top: -60px;
    transform: translateX(-50%);
    left: 50%;
    font-size:18px;
    width: auto;
    animation: fadeIn .5s ease-in forwards, fadeOut 1s ease-in-out 3s forwards;
    &--warning {
      background-color: rgb(245, 124, 0);
    }
    &--success {
      background-color: rgb(56, 142, 60);
    }
    &--info {
      background-color: rgb(2, 136, 209);
    }
    &__icon {
      font-size: 24px;
      margin-right: 5px;
    }
    @keyframes fadeIn {
      100% {
        top: 10px;
      }
    }
    @keyframes fadeOut {
      100% {
        opacity: 0;
      }
    }
  }