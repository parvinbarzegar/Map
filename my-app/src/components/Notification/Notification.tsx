import React from "react";
import cls from "classnames";
import "./Notification.scss";
import { Icon } from "@iconify/react";

interface INotificationProps {
  type?: "info" | "warning" | "success";
  message: string;
}
const Notification: React.FunctionComponent<INotificationProps> = ({
  type = "info",
  message,
}) => (
  <div
    className={cls(
      "notification",
      { "notification--info": type === "info" },
      { "notification--warning": type === "warning" },
      { "notification--success": type === "success" }
    )}
  >
    <div className=" notification__icon">
      {type === "info" && <Icon icon="codicon:info" />}
      {type === "warning" && <Icon icon="clarity:warning-line" />}
      {type === "success" && <Icon icon="bx:bx-error-circle" />}
    </div>
    {`${message}`}
  </div>
);

export default Notification;
