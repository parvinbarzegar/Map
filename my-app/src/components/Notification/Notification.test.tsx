import React from 'react';
import { screen , render, cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Notification from './Notification';

describe('Test suits for <Notification/>',()=> {
    beforeEach(()=> {
        cleanup();
    })
    it('should match to snapshot',()=> {
        const { container } = render(<Notification type="info" message='info'/>);
        expect(container).toMatchSnapshot();
    })
    it('should render a text',()=> {
        render(<Notification type="info" message='info'/>);
        const div = screen.getByText(/info/)
        expect(div).toHaveTextContent('info');
    })
    it('should be a warning notiication',()=> {
        render(<Notification type="warning" message='It is warning'/>)
        const div = screen.getByText(/It/)
        expect(div).toHaveClass('notification--warning');
    })
    it('should be a warning notiication',()=> {
        render(<Notification type="success" message='success'/>)
        const div = screen.getByText(/success/)
        expect(div).toHaveClass('notification--success');
    })
    it('should be a warning notiication',()=> {
        render(<Notification type="info" message='info'/>)
        const div = screen.getByText(/info/)
        expect(div).toHaveClass('notification--info');
    })

})