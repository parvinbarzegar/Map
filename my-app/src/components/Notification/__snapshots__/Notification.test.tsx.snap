// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for <Notification/> should match to snapshot 1`] = `
<div>
  <div
    class="notification notification--info"
  >
    <div
      class="notification__icon"
    >
      <span />
    </div>
    info
  </div>
</div>
`;
